src/main/java/com/atlassian/db/replica/api/reason/RouteDecision.java

```diff
@@ -10,11 +10,14 @@
     private final Reason reason;
     private final String sql;
     private final RouteDecision cause;
+    private final Boolean isSqlWrite;
 
-    public RouteDecision(String sql, Reason reason, RouteDecision cause) {
+    public RouteDecision(String sql, Reason reason, RouteDecision cause, Boolean isSqlWrite) {
         this.sql = sql;
         this.reason = reason;
         this.cause = cause;
+        this.isSqlWrite = isSqlWrite;
```

Comment:
It can be null. Does it mean it's an optional parameter? Why it's nullable? Can we refactor it to use a builder instead of a constructor?

Reply:
I thought if `Optional<String> getSql()` then it make sense to have isSqlWrite Optional too.
I'll  left it as nullable Boolean, as it's nullable. 

> Can we refactor it to use a builder instead of a constructor?
I don't see how this might help with nullAbility.
Builder will pass null and all will work.

Reply:
> I don't see how this might help with nullAbility.

It will not help with nullability. It will help with usability and [API evolution](https://github.com/wyrzyk/java-api-design-checklist/blob/master/CHECKLIST.md#3612-consider-a-builder-when-faced-with-many-constructor-parameters).